src/main/java/uk/gov/hmcts/reform/cpo/repository/CasePaymentOrdersRepositoryImpl.java

```diff
@@ -45,11 +44,30 @@ public void deleteByCaseIds(List<Long> caseIds) {
     }
 
     private void validateAllEntriesExistByCaseIds(List<Long> caseIds) {
+        List<String> nonExistentCaseIds = new ArrayList<>();
         for (Long cid : caseIds) {
-            if (casePaymentOrdersJpaRepository.findAllByCaseId(cid).isEmpty()) {
-                throw new CasePaymentOrderCouldNotBeFoundException(ValidationError.CPO_NOT_FOUND_BY_CASE_ID);
+            if (casePaymentOrdersJpaRepository.countAllByCaseId(cid) == 0) {
+                nonExistentCaseIds.add(String.valueOf(cid));
+            }
+        }
+        throwExceptionIfCposNotFound(nonExistentCaseIds);
+    }
+
+    private void validateAllEntriesExistByUuid(List<UUID> uuids) {
+        List<String> nonExistentUuids = new ArrayList<>();
+        for (UUID uuid : uuids) {
+            if (casePaymentOrdersJpaRepository.countAllById(uuid) == 0) {
```

Comment:
built in `casePaymentOrdersJpaRepository.existsById` is better rather than new method?

Reply:
👍